runhouse/rns/hardware/cluster.py

```diff
@@ -670,3 +673,57 @@ def remove_conda_env(
         """Remove conda env from the cluster."""
         env_name = env if isinstance(env, str) else env.env_name
         self.run([f"conda env remove -n {env_name}"])
+
+
+# Cluster factory method
+def cluster(
+    name: str,
+    ips: List[str] = None,
+    ssh_creds: Optional[dict] = None,
+    dryrun: bool = False,
+    **kwargs,
+) -> Union[Cluster, "OnDemandCluster"]:
+    """
+    Builds an instance of :class:`Cluster`.
+
+    Args:
+        name (str): Name for the cluster, to re-use later on.
+        ips (List[str], optional): List of IP addresses for the BYO cluster.
+        ssh_creds (dict, optional): Dictionary mapping SSH credentials.
+            Example: ``ssh_creds={'ssh_user': '...', 'ssh_private_key':'<path_to_key>'}``
+        dryrun (bool): Whether to create the Cluster if it doesn't exist, or load a Cluster object as a dryrun.
+            (Default: ``False``)
+
+    Returns:
+        Union[Cluster, OnDemandCluster]: The resulting cluster.
+
+    Example:
+        >>> # BYO Cluster
+        >>> gpu = rh.cluster(ips=['<ip of the cluster>'],
+        >>>                  ssh_creds={'ssh_user': '...', 'ssh_private_key':'<path_to_key>'},
+        >>>                  name='rh-a10x')
+
+        >>> # Load cluster from above
+        >>> reloaded_cluster = rh.cluster(name="rh-a10x", dryrun=True)
+    """
+    if ips is None and ssh_creds is None:
+        # If only the name is provided and dryrun is set to True
+        return Cluster.from_name(name, dryrun)
```

Comment:
I don't think this will handle the case where the name refers to an on-demand cluster